azure-pipelines-regression-test.yml

```diff
@@ -0,0 +1,309 @@
+        TestCommit: ${{ item.commit }}
+        BuildScript: ${{ item.buildScript }}
+        DisplayName: ${{ item.displayName }}
+      steps:
+      - checkout: none
+        displayName: Skip default checkout
+
+      # Download the F# compiler artifacts
+      - task: DownloadPipelineArtifact@2
+        displayName: Download F# Compiler Artifacts
+        inputs:
+          artifactName: 'FSharpCompilerArtifacts'
+          downloadPath: '$(Pipeline.Workspace)/FSharpCompiler'
+
+      - task: DownloadPipelineArtifact@2
+        displayName: Download UseLocalCompiler props
+        inputs:
+          artifactName: 'UseLocalCompilerProps'
+          downloadPath: '$(Pipeline.Workspace)/Props'
+
+      # Checkout the third-party repository at specific commit
+      - script: |
+          echo Cloning repository: $(TestRepoName)
+          git clone https://github.com/$(TestRepoName).git $(Pipeline.Workspace)/TestRepo
+          cd $(Pipeline.Workspace)/TestRepo
+          
+          echo Checking out commit: $(TestCommit)
+          git checkout $(TestCommit)
+          
+          echo Successfully checked out $(TestRepoName) at commit $(TestCommit)
+          git log -1 --oneline
+          
+          echo Repository structure:
+          dir /B
+          
+          echo Verifying build script exists: $(BuildScript)
+          if exist "$(BuildScript)" (
+            echo ✓ Build script found: $(BuildScript)
+          ) else (
+            echo ✗ Build script not found: $(BuildScript)
+            echo Available files in root:
+            dir
+            exit /b 1
+          )
+        displayName: 'Checkout $(DisplayName) at specific commit'
+
+      # Inject UseLocalCompiler.Directory.Build.props
+      - script: |
+          cd $(Pipeline.Workspace)/TestRepo
+          echo Creating Directory.Build.props to use local F# compiler
+          echo ^<Project^> > Directory.Build.props
+          echo   ^<Import Project="UseLocalCompiler.Directory.Build.props" /^> >> Directory.Build.props
+          echo ^</Project^> >> Directory.Build.props
+          
+          echo Copying UseLocalCompiler.Directory.Build.props with updated path
+          powershell -Command "& { $content = Get-Content '$(Pipeline.Workspace)/Props/UseLocalCompiler.Directory.Build.props' -Raw; $content = $content -replace 'LocalFSharpCompilerPath.*MSBuildThisFileDirectory.*', 'LocalFSharpCompilerPath>$(Pipeline.Workspace)/FSharpCompiler<'; Set-Content 'UseLocalCompiler.Directory.Build.props' -Value $content }"
+          
+          echo Directory.Build.props content:
+          type Directory.Build.props
+          echo.
+          echo UseLocalCompiler.Directory.Build.props content (first 20 lines):
+          powershell -Command "Get-Content 'UseLocalCompiler.Directory.Build.props' | Select-Object -First 20"
+        displayName: 'Setup local compiler configuration for $(DisplayName)'
+
+      # Report dotnet info in test environment
+      - script: |
```

Comment:
You have a bunch of `script` blocks laying around. Move them to tasks with their own name and display, so it gets easier to see in AzDo web UI once they finish running.